Sources/Optimizely+Decide/OptimizelyUserContext.swift

```diff
@@ -131,12 +131,53 @@ public class OptimizelyUserContext {
                                   reasons: reasonsToReport)
     }
 
-    public func decideAll(keys: [String], options: [OptimizelyDecideOption]? = nil) -> [String: OptimizelyDecision] {
-        return [:]
+    /// Returns a key-map of decision results for multiple flag keys and a user context.
+    ///
+    /// - If the SDK finds an error (__flagKeyInvalid__, etc) for a key, the response will include a decision for the key showing `reasons` for the error (regardless of __includeReasons__ in options).
+    /// - The SDK will always return key-mapped decisions. When it can not process requests (on __sdkNotReady__ or __userContextInvalid__ errors), it’ll return an empty map after logging the errors.
+    ///
+    /// - Parameters:
+    ///   - keys: An array of flag keys for which decisions will be made. When set to `nil`, the SDK will return decisions for all active flag keys.
+    ///   - options: An array of options for decision-making.
+    /// - Returns: A dictionary of all decision results, mapped by flag keys.
+    public func decideAll(keys: [String],
+                          options: [OptimizelyDecideOption]? = nil) -> [String: OptimizelyDecision] {
+        
+        guard let optimizely = self.optimizely, let _ = optimizely.config else {
+            logger.e(OptimizelyError.sdkNotReady)
+            return [:]
+        }
+        
+        guard keys.count > 0 else { return [:] }
+        
+        let allOptions = getAllOptions(with: options)
+        
+        var decisions = [String: OptimizelyDecision]()
+        
+        keys.forEach { key in
+            let decision = decide(key: key, options: options)
+            if !allOptions.contains(.enabledFlagsOnly) || decision.enabled {
+                decisions[key] = decision
+            }
+        }
+                
+        return decisions
     }
-
+    
+    /// Returns a key-map of decision results for all active flag keys.
+    ///
+    /// - Parameters:
+    ///   - options: An array of options for decision-making.
+    /// - Returns: A dictionary of all decision results, mapped by flag keys.
     public func decideAll(options: [OptimizelyDecideOption]? = nil) -> [String: OptimizelyDecision] {
-        return [:]
+        guard let optimizely = self.optimizely, let config = optimizely.config else {
```

Comment:
I think we talked about adding the guard statement in the first pr since it can be true at that point.  But, this captures the question in the first PR (error on optimizely going nil).  Do we have a unit test for this test case?